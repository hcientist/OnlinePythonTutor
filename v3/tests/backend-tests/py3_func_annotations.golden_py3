{
  "code": "# test case of Python 3 function annotations inspired by David G. Kay\ndef remove_by_name(L: 'list of Restaurant', name: str) -> 'list of Restaurant':\n    \"\"\" Given name, remove all Restaurants with that name from collection.\n    \"\"\"\n    result = []\n    for r in L:\n        if r.name != name:\n            result.append(r)\n    return result\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "remove_by_name"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "remove_by_name": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "FUNCTION", 
          "remove_by_name(L, name)", 
          null
        ]
      }, 
      "line": 2, 
      "event": "return"
    }
  ]
}
