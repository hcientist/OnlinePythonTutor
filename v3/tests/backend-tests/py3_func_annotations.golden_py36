{
  "code": "# test case of Python 3 function annotations inspired by David G. Kay\ndef remove_by_name(L: 'list of Restaurant', name: str) -> 'list of Restaurant':\n    \"\"\" Given name, remove all Restaurants with that name from collection.\n    \"\"\"\n    result = []\n    for r in L:\n        if r.name != name:\n            result.append(r)\n    return result\n",
  "trace": [
    {
      "line": 2,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 2,
      "event": "return",
      "func_name": "<module>",
      "globals": {
        "remove_by_name": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "remove_by_name"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "FUNCTION",
          "remove_by_name(L, name)",
          null
        ]
      },
      "stdout": ""
    }
  ]
}
