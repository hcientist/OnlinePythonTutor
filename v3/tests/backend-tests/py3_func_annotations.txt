# test case of Python 3 function annotations inspired by David G. Kay
def remove_by_name(L: 'list of Restaurant', name: str) -> 'list of Restaurant':
    """ Given name, remove all Restaurants with that name from collection.
    """
    result = []
    for r in L:
        if r.name != name:
            result.append(r)
    return result
