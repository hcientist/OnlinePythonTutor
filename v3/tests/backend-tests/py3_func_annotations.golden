{
  "code": "# test case of Python 3 function annotations inspired by David G. Kay\ndef remove_by_name(L: 'list of Restaurant', name: str) -> 'list of Restaurant':\n    \"\"\" Given name, remove all Restaurants with that name from collection.\n    \"\"\"\n    result = []\n    for r in L:\n        if r.name != name:\n            result.append(r)\n    return result\n", 
  "trace": [
    {
      "exception_msg": "SyntaxError: invalid syntax (<string>, line 2)", 
      "line": 2, 
      "event": "uncaught_exception", 
      "offset": 21
    }
  ]
}
